src/Umbraco.Web.UI.Client/src/common/resources/contenttype.resource.js

```diff
@@ -432,15 +432,13 @@ function contentTypeResource($q, $http, umbRequestHelper, umbDataFormatter, loca
                 throw "args.id cannot be null";
             }
 
-            var promise = localizationService.localize("contentType_moveFailed");
-
             return umbRequestHelper.resourcePromise(
                 $http.post(umbRequestHelper.getApiUrl("contentTypeApiBaseUrl", "PostMove"),
                     {
                         parentId: args.parentId,
                         id: args.id
                     }, { responseType: 'text' }),
-                promise);
+              'Failed to move content type');
```

Comment:
I saw your comment about the promise not being returned on time, but what about keeping the promise and having something like 

`promise ?? 'Failed to move content type'`

This way, if the promise ever gets returned on time, we get the foreseen label and otherwise we have a fallback.

What do you think? (same question in mediatype.resource of course😁 )

Reply:
problem is that the promise is not null but an Promise object so this won't work 🤔

There also does not seem to be an easy way of waiting for the promise to finish as there is no way to check for the state of it.

Only possibility that comes to my mind would be to have a bool which would be set to true if the promise finished in time and then set the message based on in it like so `isFinished ? promise : 'Failed to move content type'`
However by doing this we would introduce a 3rd way of handling this message as we already have the way it is done on the data types (static message) and another way for media (static message + taking some of the translation from the api response)
https://github.com/umbraco/Umbraco-CMS/blob/a0d673262ceafc636e4fc5b856b7933733e1fc23/src/Umbraco.Web.UI.Client/src/common/resources/media.resource.js#L130

Reply:
Thanks for your comments @CyberReiter !

In that case, I would suggest to keep things as they are. It's already a great improvement, so it should not hold us from merging all this 😁

Thanks again for this update 👍

Cheers!